feat: Add base64 encode decode project article (#644)
TheRenegadeCoder/sample-programs-website: sources/projects/base64/requirements.md

Write a program that accepts two parameters: the mode (encode or decode) and
some text.

If the mode is `decode`, it should print the decoded base64 text.
If the mode is `encode`, it should print the encoded text.
If the mode is neither `encode` nor `decode`, it exit with an error.

```console
$ ./base64.lang "encode" "hello world"

> aGVsbG8gd29ybGQK

Reviewer: The original encoding results in `hello world\n'. This is the encoding for `hello world`:
```suggestion
aGVsbG8gd29ybGQ=
```

Also please show an example of decode.

Author: Fixed this, good catch.

Write a program that accepts two parameters: the mode (encode or decode) and
some text.

> If the mode is `decode`, it should print the decoded base64 text.
> If the mode is `encode`, it should print the encoded text.
> If the mode is neither `encode` nor `decode`, it exit with an error.

Reviewer: ```suggestion
- If the mode is `decode`, it should print the decoded base64 text.
- If the mode is `encode`, it should print the encoded text.
- It should print an error message and exit with an error when:
  - The mode is neither `encode` or `decode`.
  - For `decode`, the length of the input string is not a multiple of 4.
```

Write a program that accepts two parameters: the mode (encode or decode) and
some text.

- If the mode is `decode`, it should print the decoded base64 text.
- If the mode is `encode`, it should print the encoded text.
- It should print an error message and exit with an error when:
  - The mode is neither `encode` or `decode`.

>   - For `decode`, the length of the input string is not a multiple of 4.

Reviewer: Sorry, forgot one more condition:
```suggestion
  - For `decode`, the length of the input string is not a multiple of 4 or contains invalid characters.
```

```console
$ ./base64.lang "encode" "hello world"
aGVsbG8gd29ybGQ=
$ ./base64.lang "decode" "aGVsbG8gd29ybGQ="
hello world

> ```

Reviewer 2: Overall, this file looks great! One thing I might add is the same caveat we added to Reverse String. Specifically, something like this: 

> Acceptable language utilities include language features and built-in libraries. External dependencies are unacceptable. Remember, the goal is to show off language features and utilities.

Because strings are so notorious for their complexity, I would probably also explicitly state the following requirement as well (note that this is in the description of reverse string, but it should probably be listed in the requirements):

> In this repository, the algorithm must handle ASCII strings. Do NOT worry about handling a string in the general case. For instance, if a string contains surrogate pairs, it's okay if the solution corrupts the string during encoding/decoding.

With that said, I am not super familiar with base64 encoding, so these issues may not actually matter.

Author: Makes sense. I've added this.
